### PR TITLE
Do not preserve EXIFIFD tag by default when saving TIFF images

### LINKED ISSUE #8106: save() produces an unreadable TIFF
`Image.open('in.tif').save('out.tif')` produces `out.tif` which is unreadable. 

`Image.__version__ == '10.1.0'`

[in.tif.zip](https://github.com/user-attachments/files/15568097/in.tif.zip)

### PR DESCRIPTION
**Pull request by radarhere**
Resolves #8106

The EXIFIFD tag is '[a pointer to the Exif IFD](https://www.awaresystems.be/imaging/tiff/tifftags/exififd.html)'. However, there's no reason why the data should be located at that very same position in the image that Pillow creates.

So this PR does not preserve it by default when saving TIFF images. The user may still provide it through an encoder argument.

### CODE REVIEW

File: Tests/test_file_libtiff.py [hunk 1]
```diff
@@ -686,12 +686,17 @@ def test_save_ycbcr(self, tmp_path: Path) -> None:
             assert reloaded.tag_v2[532] == (0, 255, 128, 255, 128, 255)
 
     def test_exif_ifd(self, tmp_path: Path) -> None:
```
Comment (Yay295) [inline Tests/test_file_libtiff.py:688]: The `tmp_path` parameter can be removed now.

File: src/PIL/TiffImagePlugin.py [hunk 2]
```diff
@@ -1819,8 +1819,9 @@ def _save(im, fp, filename):
         # SAMPLEFORMAT is determined by the image format and should not be copied
         # from legacy_ifd.
```
Comment (Yay295) [inline src/PIL/TiffImagePlugin.py:1820]: Maybe add EXIFIFD to this comment?

File: Tests/test_file_libtiff.py (hunk 1 above)
Comment (radarhere) [reply Tests/test_file_libtiff.py]: I've updated the commit.

File: src/PIL/TiffImagePlugin.py (hunk 2 above)
Comment (radarhere) [reply src/PIL/TiffImagePlugin.py]: I've updated the commit.

**Pull request merged**